portal-bridge/src/bridge/history.rs

```diff
@@ -200,11 +215,27 @@ impl HistoryBridge {
                 let epoch_acc = epoch_acc.clone();
                 let portal_clients = self.portal_clients.clone();
                 let execution_api = self.execution_api.clone();
+                let permit = match gossip_send_semaphore.clone().acquire_owned().await {
```

Comment:
If `"this should be impossible"` as you state in the error message, then I'd probably go with an `.expect("..")`